src/Serializer/JsonApiSerializer.php

```diff
@@ -358,7 +358,7 @@ protected function parseRelationships($includedData)
      */
     protected function getIdFromData(array $data)
     {
-        if (!array_key_exists('id', $data)) {
+        if (!array_key_exists('id', $data) || (!is_string($data['id']) && !is_numeric($data['id']))) {
```

Comment:
I think we'd like to accept anything that can be cast as a string. We'd want to check something along the lines of `is_scalar` or implements `__toString()`. I don't see any reason not to allow string-able ids.